doc/source/whatsnew/v2.3.0.rst

```diff
@@ -28,7 +28,15 @@ enhancement2
 
 Other enhancements
 ^^^^^^^^^^^^^^^^^^
--
+- ``pipe`` methods—such as
+  :meth:`DataFrame.pipe`,
+  :meth:`Series.pipe`,
+  :meth:`DataFrameGroupBy.pipe`,
+  :meth:`SeriesGroupBy.pipe`,
+  :meth:`Resampler.pipe`,
+  and :meth:`Styler.pipe`
+  now have type hints for the parameters
+  they pass through to their respective input functions ``func``.
```

Comment:
We typically have no whatsnew for type annotations.

Merging/rebasing might fix the doc-test failure.

Reply:
Ah ok well if this isn't something we want I'll just remove the latest commit, and that will probably be ready for review.

Reply:
Don't forget to click "ready to review" when you are ready.

The only obvious thing I noticed is that S is still defined and used in the overloads.

Reply:
> Don't forget to click "ready to review" when you are ready.

I will! Thanks for the reminder, still experimenting with a few things before it's good to go.